core/lib/types/src/storage/writes/mod.rs

```diff
@@ -188,6 +188,78 @@ fn prepend_header(compressed_state_diffs: Vec<u8>) -> Vec<u8> {
     res.to_vec()
 }
 
+/// Struct for storing tree writes in DB.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct TreeWrite {
+    /// `address` part of storage key.
+    pub address: Address,
+    /// `key` part of storage key.
+    pub key: H256,
+    /// Value written.
+    pub value: H256,
+    /// Leaf index of the slot.
+    pub leaf_index: u64,
+}
+
+impl TreeWrite {
+    const TREE_WRITE_BYTES_SIZE: usize = 20 + 32 + 32 + 8;
+
+    fn extend_with_bytes(&self, vec: &mut Vec<u8>) {
+        vec.extend_from_slice(&self.address.0);
+        vec.extend_from_slice(&self.key.0);
+        vec.extend_from_slice(&self.value.0);
+        vec.extend_from_slice(&self.leaf_index.to_be_bytes());
+    }
+
+    pub fn from_slice(slice: &[u8]) -> anyhow::Result<Self> {
```

Comment:
Nit: Does this method need to be public? It seems to be only used from `vec_from_slice`.